File: ch4/yojkim/ch4/dollar_test.go
package ch4

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestMultiplication(t *testing.T) {
	five := NewDollar(5)
	assert.Equal(t, *five.Times(2), NewDollar(10))
	assert.Equal(t, *five.Times(3), NewDollar(15))
}

func TestEquality(t *testing.T) {
	assert.True(t, NewDollar(5).Equal(NewDollar(5)))
	assert.False(t, NewDollar(5).Equal(NewDollar(6)))
}

File: ch4/gunddol/src/ch4/dollar_Test.java
package ch4;

import org.junit.Test;
import static org.junit.Assert.*;

public class dollar_Test {

    @Test
    public void testMultiplication() {
        dollar five = new dollar(5);
        assertEquals(new dollar(10), five.times(2));
        assertEquals(new dollar(15), five.times(3));
    }

    @Test
    public void testEquality(){
        assertTrue(new dollar(5).equals(new dollar(5)));
        assertFalse(new dollar(5).equals(new dollar(6)));
    }
}